### PR TITLE
Generate javadoc for XTF sources

### PR DESCRIPTION
**Pull request by mnovak1**
Public Maven repos require to have javadoc when releasing. So this is to fullfill requirements for new Maven release repo (https://github.com/xtf-cz/xtf/issues/423)

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/config/XTFConfig.java [hunk 1]
```diff
@@ -12,14 +12,16 @@
 import lombok.extern.slf4j.Slf4j;
 
 /**
- * Loads properties stored in several different ways. Possible options, with top to down overriding are: </br>
+ * Loads properties stored in several different ways. Possible options, with top to down overriding are: <br />
+ *
```
Comment (simkam) [inline core/src/main/java/cz/xtf/core/config/XTFConfig.java:16]: why additional new line?

File: core/src/main/java/cz/xtf/core/config/XTFConfig.java [hunk 2]
```diff
@@ -12,14 +12,16 @@
...
  *
  * <ul>
  * <li><i>global-test.properties</i> on project root path</li>
  * <li><i>test.properties</i> on project root path - meant to be user specific, unshared</li>
  * <li><i>environment variables</i> in System.getEnv()</li>
  * <li><i>system properties</i> in System.getProperties()</li>
  * </ul>
+ *
```
Comment (simkam) [inline core/src/main/java/cz/xtf/core/config/XTFConfig.java:24]: new line?

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 3]
```diff
@@ -828,6 +834,8 @@ public BuildConfig updateBuildConfig(BuildConfig buildConfig) {
      *
      * @param name name of buildConfig
      * @param envVars environment variables
+     *
+     * @return returns build config
```
Comment (simkam) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java:838]: `@returns` are incosistent, `ProjectRequest` / `deployment config` / `returns build config`

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 4]
```diff
@@ -261,7 +262,9 @@ public ProjectRequest createProjectRequest(ProjectRequest projectRequest) {
     /**
      * Calls recreateProject(namespace).
      *
+     * @return project request
      * @see OpenShift#recreateProject(String)
+     *
```
Comment (simkam) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java:267]: new line?

**Review by simkam — CHANGES_REQUESTED** [review]

File: core/src/main/java/cz/xtf/core/config/XTFConfig.java (hunk 1 above)
Comment (mnovak1) [reply core/src/main/java/cz/xtf/core/config/XTFConfig.java:16]: fixed

File: core/src/main/java/cz/xtf/core/config/XTFConfig.java (hunk 2 above)
Comment (mnovak1) [reply core/src/main/java/cz/xtf/core/config/XTFConfig.java:24]: fixed

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 4 above)
Comment (mnovak1) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java:267]: fixed

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 3 above)
Comment (mnovak1) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java:838]: I went through the `@returns` and fixed those.

**Comment by mnovak1** [comment]
@simkam I've fixed the issues, could you check it again?

**Review by simkam — APPROVED** [review]

**Pull request merged**